### PR TITLE
Task.Run -> ThreadPool.QueueUserWorkItem

### PR DESCRIPTION
**Pull request by benaadams**
Task.Run eventually ends up being QueueUserWorkItem.
The returned task is ignored, so no added goodness.
Short running item.

Cut out the middleman

### CODE REVIEW

File: src/Microsoft.AspNet.Server.Kestrel/Http/SocketInput.cs [hunk 1]
```diff
@@ -194,7 +190,7 @@ public void OnCompleted(Action continuation)
             }
             else if (awaitableState == _awaitableIsCompleted)
             {
-                Task.Run(continuation);
+                ThreadPool.QueueUserWorkItem((o) => continuation());
```
Comment (rynowak) [inline src/Microsoft.AspNet.Server.Kestrel/Http/SocketInput.cs]: Should this become an `Action<object>`? Isn't this going to allocate a lambda?

Comment (benaadams) [inline src/Microsoft.AspNet.Server.Kestrel/Http/SocketInput.cs]: Gah, yeah. Changed

**Comment by benaadams** [comment]
Closing for now until have some numbers

**Comment by benaadams** [comment]
There is a flag switch that differs between the two paths (`compressedStack`); will need to evaluate more.

File: src/Microsoft.AspNet.Server.Kestrel/Http/SocketInput.cs [hunk 2]
```diff
@@ -97,11 +97,7 @@ public void IncomingComplete(int count, Exception error)
                     {
                         _head = _tail = _pinned;
                     }
-                    else if (_tail == _pinned)
```
Comment (davidfowl) [inline src/Microsoft.AspNet.Server.Kestrel/Http/SocketInput.cs]: avoid the unrelated changes in PR, they're pretty distracting.
